lib/cocoapods/bazel/target.rb

```diff
@@ -473,8 +473,9 @@ def deps_by_config
         labels_by_config = {}
 
         if !sorted_debug_labels.empty? || !sorted_release_labels.empty?
-          labels_by_config[build_settings_label(:debug)] = sorted_debug_labels
-          labels_by_config[build_settings_label(:release)] = sorted_release_labels
```

Comment:
It seems this changes the default behavior for release? Was that an expected change? I saw the spec change but I wasn't sure if that'd break existing uses. 

Any downsides to having an additional permutation that enables `combin_release_and_debug_deps` on just the deps ATTR but leaves the others?

Reply:
I'm currently trying to figure out an alternative implementation to avoid introduction a breaking change. You're right, this would change the current behaviour for `deps`. The problem I'm facing is, if I created a combine setting say `debug_and_release` int the `BUILD` file we'd have:
```
    deps = select({
        "//Pods/cocoapods-bazel:debug": ["//some/debug/dep"],
        "//Pods/cocoapods-bazel:release": ["//some/release/dep"],
        "//Pods/cocoapods-bazel:debug_and_release": ["//some/debug/dep", "//some/release/dep"],
    }) 
```
that works fine if I build with `--//Pods/cocoapods-bazel:config=debug_and_release`, but since the same config name is also used for other attributes (not `swift_copts`, `objc_copts`, etc) we also want to continue to be able to specify `--//Pods/cocoapods-bazel:config=release` in the command line. This breaks the select for `deps` above because bazel doesn't know which list of `deps` to select with both settings set in the command line.

In the use case I mentioned we don't want to apply `debug_and_release` everywhere, but only for `deps` instead, this is similar to say `%w[Debug Release]` in a `.podspec`

Reply:
Yeah that's what I was seeing as well

Reply:
https://github.com/bazel-ios/cocoapods-bazel/pull/45#issuecomment-925935675